python/ray/train/predictor.py

```diff
@@ -124,6 +123,19 @@ def set_preprocessor(self, preprocessor: Optional[Preprocessor]) -> None:
         """Set the preprocessor to use prior to executing predictions."""
         self._preprocessor = preprocessor
 
+    @classmethod
+    def preferred_batch_format(cls) -> BatchFormat:
```

Comment:
using class method is easier imo since in `BatchPredictor` all we got is the class of predictor rather than instance, thus it's accessed by `self._predictor_class. preferred_batch_format()` without being leaky abstraction

Reply:
> are there any immediate use cases for a predictor determining its preferred batch format based off of instance data

Not that i'm aware of, all DLPredictor only works with numpy thus tabular format always incurs higher cost. I think you're implying another symptom that user provided a bunch of preprocessors that could make hard conversion inevitable (like read arrow -> pandas UDF -> numpy UDF -> TorchPredictor) and in that case I think we should aim for making writing efficient UDFs easier, more flexible and better integrated with dataset plan optimizer rather than within consumer itself.